libiop/bcs/hashing/blake2b.cpp

```diff
@@ -47,6 +49,33 @@ binary_hash_digest blake2b_two_to_one_hash(const binary_hash_digest &first,
     return result;
 }
 
+binary_hash_digest blake2b_many_to_one_hash(const std::vector<binary_hash_digest> &data,
+                                            const std::size_t digest_len_bytes)
+{
+    /* binary_hash_digest is a C++ string pointer so we need to sum them since they are not
+       contiguous in memory. */
+    binary_hash_digest input = data[0];
+    for (int i = 1; i < data.size(); i++)
+    {
+        input += data[i];
+    }
+
+    binary_hash_digest result(digest_len_bytes, 'X');
+
+    /* see https://download.libsodium.org/doc/hashing/generic_hashing.html */
+    const int status = crypto_generichash_blake2b((unsigned char*)&result[0],
+                                                  digest_len_bytes,
+                                                  (result.empty() ? NULL : (unsigned char*)&input[0]),
```

Comment:
should it be `result.empty()` or `input.empty()`?

Reply:
Should be input